lib/lbt/bundle/Resolver.js

```diff
@@ -56,8 +56,16 @@ class BundleResolver {
 				return match;
 			}
 
-			function isMultiModule(moduleInfo) {
-				return moduleInfo && moduleInfo.subModules.length > 0 && !/(?:^|\/)library.js$/.test(moduleInfo.name);
+			function isDecomposableBundle(moduleInfo) {
+				if ( moduleInfo == null
+						|| moduleInfo.subModules.length === 0
+						|| /(?:^|\/)library.js$/.test(moduleInfo.name) ) {
+					return false;
+				}
+				// it might look more natural to expect 'all' embedded modules to exist in the pool,
+				// but expecting only one module is a more conservative approach
+				return Promise.all(moduleInfo.subModules.map((sub) => pool.findResource(sub).catch(() => false)))
```

Comment:
As already [mentioned by @kristian](https://github.com/SAP/ui5-builder/commit/265e5b64ea521993628e311ca3f27b6985692189#r35226030) I think this won't work. The line is also not covered by any test.

Reply:
Yes, you're both totally right.